Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_To_Data_Processing_Registration.sql

```diff
@@ -0,0 +1,622 @@
+﻿/*
+
+Content:
+ Migrates existing system and contract GDPR data to Data Processing Registrations according to the logic described in https://os2web.atlassian.net/browse/KITOSUDV-1161
+
+*/
+
+
+/*
+	First migration situation
+	Contract with ContractType as "Databehandleraftale" (5) and 1 to many associated it system usages
+*/
+
+BEGIN
+
+	DECLARE @MigrationContext1 TABLE 
+    (
+		rowNumber int IDENTITY(1,1) PRIMARY KEY,
+		ItContractId int,
+		ItSystemUsageId int,
+		OrganizationId int,
+		LastChanged datetime2(7),
+		Name varchar(max),
+		AgreementConcluded int, 
+		AgreementConcludedUrl varchar(max),
+		AgreementConcludedUrlName varchar(max),
+		DataResponsibleNote varchar(max),
+		OversightConcluded int,
+		OversightConcludedDate varchar(max),
+		OversighConcludedRemark varchar(max),
+		DatahandlerSupervisionLink varchar(max),
+		DatahandlerSupervisionLinkName varchar(max),
+		ContractUserId int,
+		ContractLastChangedBy int,
+		SystemUsageUserId int
+    )
+	
+
+	/*
+		Get all info needed for first migration situation
+	*/
+
+	INSERT INTO @MigrationContext1 (ItContractId, ItSystemUsageId, OrganizationId, LastChanged, Name, AgreementConcluded, AgreementConcludedUrl, AgreementConcludedUrlName, DataResponsibleNote, OversightConcluded, OversightConcludedDate, OversighConcludedRemark, DatahandlerSupervisionLink, DatahandlerSupervisionLinkName, ContractUserId, ContractLastChangedBy, SystemUsageUserId)
+        SELECT
+			ItContractItSystemUsages.ItContractId, 
+			ItContractItSystemUsages.ItSystemUsageId,
+			ItContract.OrganizationId,
+			GETUTCDATE(),
+			'Konverteret_Databehandling_',
+			ItContract.ContainsDataHandlerAgreement,
+			ItContract.DataHandlerAgreementUrl,
+			ItContract.DataHandlerAgreementUrlName,
+			ItSystemUsage.dataProcessor,
+			CASE
+				WHEN ItSystemUsage.dataProcessorControl = 0 THEN 1
+				WHEN ItSystemUsage.dataProcessorControl = 1 THEN 0
+				WHEN ItSystemUsage.dataProcessorControl IS NULL THEN NULL
+				ELSE 2
+			END AS OversightConcluded,
+			ItSystemUsage.lastControl,
+			ItSystemUsage.noteUsage,
+			ItSystemUsage.datahandlerSupervisionDocumentationUrl,
+			ItSystemUsage.datahandlerSupervisionDocumentationUrlName,
+			ItContract.ObjectOwnerId,
+			ItContract.LastChangedByUserId,
+			ItSystemUsage.ObjectOwnerId
+        FROM 
+		ItContractItSystemUsages 
+			INNER JOIN ItContract ON ItContract.Id = ItContractItSystemUsages.ItContractId
```

Comment:
Du kan godt teste begge conditions med det samme i stedet for en ekstra join: https://github.com/Strongminds/kitos/commit/448e408bf785b11dc99d39c8f949dac105fda30c#diff-5c93e69623acf497c29a0e7bdf4008f28a332ac130c39bc50152221e8dcef201R71

Reply:
Det er to INNER JOIN's jeg laver. Så jeg skal samle alle 3 tabeller (2 joins) for at få min samlede tabel da jeg skal bruge data fra både ItContract og ItSystemUsage